src/main/java/com/minecolonies/util/BlockUtils.java

```diff
@@ -28,19 +29,42 @@
     @NotNull
     private static List<BiPredicate<Block, IBlockState>> freeToPlaceBlocks =
       Arrays.asList(
-        (block, iBlockState) -> block.equals(Blocks.air),
-        (block, iBlockState) -> block.getMaterial().isLiquid(),
+        (block, iBlockState) -> block.equals(Blocks.AIR),
+        (block, iBlockState) -> iBlockState.getMaterial().isLiquid(),
         (block, iBlockState) -> BlockUtils.isWater(block.getDefaultState()),
-        (block, iBlockState) -> block.equals(Blocks.leaves),
-        (block, iBlockState) -> block.equals(Blocks.leaves2),
-        (block, iBlockState) -> block.equals(Blocks.double_plant),
-        (block, iBlockState) -> block.equals(Blocks.grass),
+        (block, iBlockState) -> block.equals(Blocks.LEAVES),
+        (block, iBlockState) -> block.equals(Blocks.LEAVES2),
+        (block, iBlockState) -> block.equals(Blocks.DOUBLE_PLANT),
+        (block, iBlockState) -> block.equals(Blocks.GRASS),
         (block, iBlockState) -> block instanceof BlockDoor
                                   && iBlockState != null
                                   && iBlockState.getValue(PropertyBool.create("upper"))
 
       );
 
+
+    /**
+     * Updates the rotation of the structure depending on the input.
+     *
+     * @param rotation the rotation to be set.
+     * @return returns the Rotation object.
+     */
+    public static Rotation getRotation(final int rotation)
+    {
+        switch (rotation)
+        {
+            case 1:
+                return Rotation.CLOCKWISE_90;
+            case 2:
+                return Rotation.CLOCKWISE_180;
+            case 3:
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)